packages/react-dom/src/__tests__/ReactErrorBoundaries-test.internal.js

```diff
@@ -839,19 +938,33 @@ describe('ReactErrorBoundaries', () => {
     expect(container.firstChild.textContent).toBe('Caught an error: Hello.');
   });
 
-  it('mounts the error message if mounting fails', () => {
+  it('mounts the error message if mounting fails', async () => {
     function renderError(error) {
       return <ErrorMessage message={error.message} />;
     }
 
     const container = document.createElement('div');
-    ReactDOM.render(
-      <ErrorBoundary renderError={renderError}>
-        <BrokenRender />
-      </ErrorBoundary>,
-      container,
-    );
+    const root = ReactDOMClient.createRoot(container);
+    await act(async () => {
+      root.render(
+        <ErrorBoundary renderError={renderError}>
+          <BrokenRender />
+        </ErrorBoundary>,
+      );
+    });
     assertLog([
+      'ErrorBoundary constructor',
```

Comment:
I added these expected log statements to make the tests pass I know, `createRoot(..).render` is more efficient than `ReactDOM.render` so I assumed this is a byproduct of these functional differences but that could be wrong.